resolver-dns/src/test/java/io/netty/resolver/dns/DnsNameResolverTest.java

```diff
@@ -382,16 +383,17 @@ public class DnsNameResolverTest {
     private static final TestDnsServer dnsServer = new TestDnsServer(DOMAINS_ALL);
     private static final EventLoopGroup group = new NioEventLoopGroup(1);
 
-    private static DnsNameResolverBuilder newResolver(boolean decodeToUnicode) {
-        return newResolver(decodeToUnicode, null);
+    private static DnsNameResolverBuilder newResolver(DnsNameResolverChannelStrategy strategy,
```

Comment:
In my minds eye we can (should?) run all tests with each strategy. Instead of making nearly all tests parameterized, could we instead make the test suite abstract and pass in a strategy as part of the constructor, then make two a new concrete class for each strategy that passes in the correct enum value?

That would require the `private static DnsResolverType newResolver(..)` family of methods to no longer be static but that seems like a smaller changeset than this.

Reply:
This is not possible with Junit5 :( https://github.com/junit-team/junit5/issues/878